test/oraclize/OracleHubOraclize.js

```diff
@@ -136,4 +155,159 @@ contract('OracleHubOraclize', function(accounts) {
+
+    const waitForUpdatedLastPriceEvent = ms =>
+      new Promise((resolve, reject) => {
+        const check = () => {
+          if (txReceipt) resolve();
+          else if ((ms -= 1000) < 0) reject(new Error('Oraclize time out!'));
+          else {
+            setTimeout(check, 1000);
+          }
+        };
+        setTimeout(check, 1000);
+      });
+
+    await waitForUpdatedLastPriceEvent(60000);
+    assert.notEqual(
+      txReceipt,
+      null,
+      'Oraclize callback did not inovke marketContract.oracleCallback()'
+    );
+  });
+
+  it('gas used by OracleHub callback is within specified limit', async function() {
+    const nowInSeconds = Math.floor(Date.now() / 1000);
+    const marketContractExpirationInThreeSeconds = nowInSeconds + 3;
+    const priceFloor = 1;
+    const priceCap = 150;
+    let gasLimit = 6500000; // gas limit for development network
+    let block = web3.eth.getBlock('latest');
+    if (block.gasLimit > 7000000) {
+      // coverage network
+      gasLimit = block.gasLimit;
+    }
+    let txReceipt = null; // Oraclize callback transaction receipt
+
+    const deployedMarketContract = await MarketContractOraclize.new(
+      'ETHUSD-10',
+      [accounts[0], collateralToken.address, collateralPool.address],
+      oracleHub.address,
+      [priceFloor, priceCap, 2, 2, marketContractExpirationInThreeSeconds],
+      oracleDataSource,
+      oracleQuery
+    );
+
+    // request query update
+    await oracleHub.setFactoryAddress(accounts[1], { from: accounts[0] });
+    await oracleHub.requestQuery(deployedMarketContract.address, oracleDataSource, oracleQuery, 0, {
+      from: accounts[1]
+    });
+
+    let oraclizeCallbackGasCost = await oracleHub.QUERY_CALLBACK_GAS.call();
+    let contractSettledEvent = deployedMarketContract.ContractSettled();
+
+    contractSettledEvent.watch(async (err, response) => {
+      if (err) {
+        console.log(err);
+      }
+      assert.equal(response.event, 'ContractSettled');
+      txReceipt = await web3.eth.getTransactionReceipt(response.transactionHash);
+      // console.log('Oraclize callback gas used : ' + txReceipt.gasUsed);
+      assert.isBelow(
+        txReceipt.gasUsed,
+        oraclizeCallbackGasCost.toNumber(),
+        'Callback tx claims to have used more gas than allowed!'
+      );
+      contractSettledEvent.stopWatching();
+    });
+
+    const waitForContractSettledEvent = ms =>
+      new Promise((resolve, reject) => {
+        const check = () => {
+          if (txReceipt) resolve();
+          else if ((ms -= 1000) < 0) reject(new Error('Oraclize time out!'));
+          else {
+            setTimeout(check, 1000);
+          }
+        };
+        setTimeout(check, 1000);
+      });
+
+    await waitForContractSettledEvent(60000);
+    assert.notEqual(
+      txReceipt,
+      null,
+      'Oraclize callback did not arrive. Please increase QUERY_CALLBACK_GAS!'
+    );
+  });
+
+  describe('requestOnDemandQuery()', function() {
```

Comment:
@perfectmak I may not be reading the blow code correctly, but are we testing that the query gets requested or in fact that the query gets requested, succeeds, and then calls a MarketContract back?

Reply:
Yeah, I'm only testing to ensure the appropriate event is emitted.

To actually test the callback, I'll need to set the query through `requestQuery` before performing an on demand query and then check for the `UpdatedLastPrice` event. 
The problem I had was that I was getting two `UpdatedLastPrice` events within the test period (one from the request and the the other from the on demand). So I felt to remove that test, since it might end up being similar to the normal request query tests done before.